### DISCUSSION TITLE
OTBR: pointers for running tests

### DISCUSSION THREAD

**Discussion by tleacmcsa** [openthread/openthread#10928] [Q&A] [answered]
I am going to make some enhancements to the dbus interface to otbr-agent and need to add and run the related tests.  I'm new to development on this project, so I'm looking for the recommended strategy. There are many build and runtime dependencies for this stuff so I was hoping to find a docker environment that had everything required to build and run these tests but I must be missing it.  Pointers appreciated.

**Comment by superwhd** [comment, marked as answer]
As far as I know there isn't such a docker environment maintained by openthread for **developing** openthread projects. You may consider building a container which installs the dependencies like https://github.com/openthread/ot-br-posix/blob/main/script/bootstrap.

On the other hand, there are docker containers used for running the multi-BR test cases. The container may be useful for developing OTBR, if some necessary changes are applied. See [this script](https://github.com/openthread/openthread/blob/3702c88a76486fa04448e578158c25390a9c6599/script/test#L338) for how it builds the container.

**Reply by tleacmcsa** [reply]
Thanks!
